sklearn/model_selection/_search.py

```diff
@@ -732,7 +732,79 @@ def _select_best_index(refit, refit_metric, results):
             if best_index < 0 or best_index >= len(results["params"]):
                 raise IndexError("best_index_ index out of range")
         else:
-            best_index = results[f"rank_test_{refit_metric}"].argmin()
+            # Compute Rank Score
+            rank_test_score = results[f"rank_test_{refit_metric}"]
+
+            # Extract indices with minimum rank
+            min_rank = min(rank_test_score)
+            min_rank_indices = [
+                i for i, rank in enumerate(rank_test_score) if rank == min_rank
+            ]
+
+            # Check if there is only one minimum rank, and not multiple min ranks
+            if len(min_rank_indices) == 1:
+                # Store the only best index, further computation need not be necessary
+                best_index = min_rank_indices[0]
```

Comment:
I think you might need a early return here if further computation is not needed or else best_index will be overwritten?